### ISSUE TITLE
Creating a Spring Starter project in non-default file system location

### ISSUE THREAD

**Issue by fenjen** [spring-projects/sts4#1025]
**Description**

When creating a Spring Starter project in a non-default, custom file system location, then Eclipse/STS doesn't create the project correctly.

**Steps to Reproduce**
1. New Spring Starter Project
2. Name: `MyProject`
3. Uncheck `[x] Use default location` and change the path
4. Set Artifact to `artifact.id`
5. Add project to working set: `My Working Set`
6. Complete the form
7. Check the Package Explorer view.

_Expected_
```
Other Projects
My Working Set
    - MyProject (open)
```

_Actual_
```
Other Projects
    - artifact.id
My Working Set
    - MyProject (closed)
```

**Additional Information**
- Windows 11
- Eclipse 2023-03 (4.27.0)
- STS 4.18
- There are no new log messages in `workspace/.metadata/.log` when creating the project
- "Run as Spring Boot App" on `artifact.id` seems to work.
- The closed project `MyProject` cannot be deleted. It disappears after restarting Eclipse.
- Creating Spring Starter projects in default location works as expected.
- Creating a Java or Maven project in non-default locations works as expected.

**Workaround**
1. Create Spring Starter Project with [x] use default location checked.
8. Exit Eclipse.
9. Move project directory to non-default location on command line.
10. Delete the project in Eclipse and re-import it using Import - "Existing Projects into Workspace".

**Comment by BoykoAlex** [comment]
I cannot reproduce this in Win 10 with STS 4.18.0 :-\
Once a project is created in the non-default location on the file system, did you see the contents of the created folder? Any thing in it? Can you import this folder as "Existing Eclipse Project" or "Existing Maven Project"

Are you able to create a simple Java  project in the non-default location? What about simple project (General -> Project)?

**Comment by BoykoAlex** [comment]
@fenjen I should have paid more attention to the description :-) I was able to reproduce the issue. However, the expected behaviour for the project in non-default location (i.e. not under the workspace root) is going to be:
```
Other Projects
My Working Set
    - artifact.id (open)
```
(not `MyProject`)
This is due to the way m2e works. Project name is always artifact id with the exception of project located under the workspace root when the name of the eclipse project is the name (i.e. label) of the Maven project. Therefore, the fix is let M2E or Buildship create the project and then the resultant project is added to the working set.

**Comment by BoykoAlex** [comment]
Fixed with 0694331fdc8683390036ffc9f38d3a9b5f7833e6
